demo-extended/templates/deployment.yaml

```diff
@@ -31,18 +34,38 @@ spec:
           {{- end }}
           {{- end }}
           {{- if or ((.Values.scaling).resources).memory ((.Values.scaling).resources).cpu }}
+
           resources:
-            {{- if ((.Values.scaling).resources).memory }}
+            {{- if .Values.scaling.resources.memory }}
             limits:
-              memory: {{ ((.Values.scaling).resources).memory }}Gi
+              memory: "{{ .Values.scaling.resources.memory }}Gi"
             {{- end }}
-            {{- if or ((.Values.scaling).resources).memory ((.Values.scaling).resources).cpu }}
+            {{- if .Values.scaling.resources.cpu }}
             requests:
-              {{- if ((.Values.scaling).resources).cpu }}
-              cpu: {{ ((.Values.scaling).resources).cpu }}
-              {{- end }}
-              {{- if ((.Values.scaling).resources).memory }}
-              memory: {{ ((.Values.scaling).resources).memory }}Gi
-              {{- end }}
+              cpu: "{{ .Values.scaling.resources.cpu }}"
             {{- end }}
+
+          {{- if .Values.probes.enabled }}
```

Comment:
Can you add brackets the same way it was done in the resources object to avoid null objects